### ISSUE TITLE
Warning GC27A339F: method 'collectionView:numberOfItemsInSection:' in protocol 'UICollectionViewDataSource'

### ISSUE THREAD

**Issue by RenjitGit** [MvvmCross/MvvmCross#3979]
Following warning is occurring in the iOS Project while building Xamarin.iOS (xamarin native application)

`TestApp.iOS/obj/iPhone/Debug/device-builds/iphone10.2-11.4/mtouch-cache/registrar.m(17,17): Warning GC27A339F: method 'collectionView:numberOfItemsInSection:' in protocol 'UICollectionViewDataSource' not implemented [-Wprotocol] (GC27A339F) (TestApp.iOS)`

When double click (navigate to details) on the waring navigates to following auto generated code.

```
@implementation MvvmCross_Platforms_Ios_Binding_Views_MvxBaseCollectionViewSource {
	XamarinObject __monoObjectGCHandle;
}

```
Expected behavior:
Warning should not come

Reproduction steps
Build the iOS application and check the Source -> Warnings

Configuration
Debug and Release

Version:
MVVMCross 6.4.2

Platform:
iphone: iOS - 11

Environment:
Visual studio for mac 8.7.8 (build 4)
Xamarin.iOS - Version: 14.0.0.0 (Visual Studio Community)
Mono 6.12.0.93 

Please let us know, whether this warning will be fixed in the upcoming release?
Whether this will cause any issue for iOS compliance?

**Comment by Cheesebaron** [comment]
Upcoming release? Only if someone looks at the issue 😀

This shouldn't cause any issues.

**Comment by RenjitGit** [comment]
Hi @Cheesebaron,
Could you please let me know, in which MVVMCross release this fix will be added?

**Comment by Cheesebaron** [comment]
@RenjitGit next version, but don't know when it will be released.

You still haven't explained what this compliance you are talking about is and why it is important to you.

**Comment by RenjitGit** [comment]
Okay @Cheesebaron, Thanks.
As per our application requirement, We should not have warnings in our application.
iOS compliance, it's about will apple reject the application while app store review process or due to this warning application may have bugs.

**Comment by Cheesebaron** [comment]
1. It won't get rejected by Apple.
2. It is not a MSBuild warning, so not sure how you will catch it unless you are parsing logs for your builds.

**Comment by RenjitGit** [comment]
Hi @Cheesebaron,
Okay Thanks for the update.
This warning occurs when I build iOS application and able to see this warning in Errors window of Visual studio for Mac.

Hope this code update is already merged part of mvvmross-8.0.0-release (Jun 11, 2021). Please confirm.
